plan/util.go

```diff
@@ -45,3 +46,60 @@ func (a *AggregateFuncExtractor) Leave(n ast.Node) (ast.Node, bool) {
 	}
 	return n, true
 }
+
+// logicalSchemaProducer stores the schema for the logical plans who can produce schema directly.
+type logicalSchemaProducer struct {
+	schema *expression.Schema
+	baseLogicalPlan
+}
+
+// Schema implements the Plan.Schema interface.
+func (s *logicalSchemaProducer) Schema() *expression.Schema {
+	if s.schema == nil {
+		s.schema = expression.NewSchema()
+	}
+	return s.schema
+}
+
+// SetSchema implements the Plan.SetSchema interface.
+func (s *logicalSchemaProducer) SetSchema(schema *expression.Schema) {
```

Comment:
Can we only some special logical plans have this function？
I think it is more readable.